### PR TITLE
Fix rendering 1st ten nodes; refactor someDataLoaded as a method...

### PR DESCRIPTION
**Pull request by sloria**
...rather than a prop, since it is derived data

### CODE REVIEW

**Comment by caneruguz** [comment]
@pattisdr can you review this too?

**Comment by pattisdr** [comment]
Yes, we are discussing this.

**Comment by pattisdr** [comment]
@caneruguz, I thought you said I should use an m.prop() for the spinner. Can someone explain to me why I don't need it?

**Comment by pattisdr** [comment]
@This change breaks rendering of new and noteworthy: EDIT: I think it is the new redraws on New and Noteworthy.

screen shot 2016-03-10 at 4 36 55 pm

File: website/static/js/home-page/newAndNoteworthyPlugin.js [hunk 1]
```diff
@@ -56,7 +61,10 @@ var NewAndNoteworthy = {
                 self.popularNodes().push(result.data[l]);
                 self.fetchContributors(result.data[l]);
             }
-            self.someDataLoaded(true);
+            // NOTE: This manual redraw is necessary because we set background: true on
+            // the request, which prevents a redraw. This redraw allows the loading
+            // indicator to go away and the first 10 nodes to be rendered
+            m.redraw();
```
Comment (pattisdr) [inline website/static/js/home-page/newAndNoteworthyPlugin.js]: I would remove.  Discover Public Projects are only two requests.  One to fetch new and noteworthy and one to fetch most popular.   Removing these prevents the javascript error.

File: website/static/js/home-page/newAndNoteworthyPlugin.js [hunk 2]
```diff
@@ -41,7 +43,10 @@ var NewAndNoteworthy = {
                 self.newAndNoteworthyNodes().push(result.data[l]);
                 self.fetchContributors(result.data[l]);
             }
-            self.someDataLoaded(true);
+            // NOTE: This manual redraw is necessary because we set background: true on
+            // the request, which prevents a redraw. This redraw allows the loading
+            // indicator to go away and the first 10 nodes to be rendered
+            m.redraw();
```
Comment (pattisdr) [inline website/static/js/home-page/newAndNoteworthyPlugin.js]: Remove the m.redraw()?

**Comment by pattisdr** [comment]
Also, the m.redraw() on QuickSearch prevents the load more button from showing until all projects are loaded.  Original behavior was the user could load projects as long as there were projects to be reloaded.

(This happens when new and noteworthy nodes are populated. I am not sure of the behavior if you haven't done this.)

**Pull request merged**